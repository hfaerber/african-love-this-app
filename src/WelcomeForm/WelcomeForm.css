.section-radio {
  width: 45%;
  text-align: left;
}

.label-radio {
  display: block;
  margin-left: 6%;
  margin-top: 3%;
}

.button-radio {
  margin: 22px 0px 4px 0px;
  cursor: pointer;
}

.input-name {
  font-size: 1em;
  margin: 4%;
  border-radius: 3px;
  padding-left: 5px;
}

input[type='radio'] {
  transform: scale(.9);
  float: left;
}

.welcome {
  font-size: 2em;
  margin: 5% 0 2% 0;
  color: #f28242;
}

.section-instructions{
  width: 45%;
  text-align: left;
}

.title-welcome-info {
  font-size: 1.3em;
  color: #f28242;
}

.div-welcome-page-buttons {
  width: 29%;
  margin: auto;
  display: flex;
  justify-content: space-around;
}

.div-welcome-info {
  display: flex;
  padding: 3%;
  justify-content: space-between;
}

.list-instructions {
  list-style-type: square;
}

li {
  margin-top: 3%;
}
