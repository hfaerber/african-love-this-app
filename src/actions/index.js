export const updateUser = user => ({
  type: 'UPDATE_USER',
  user
});

export const updateCountries = countries => ({
  type: 'UPDATE_COUNTRIES',
  countries
});

export const toggleStudyStatus = name => ({
  type: 'TOGGLE_STUDY_STATUS',
  name
});

export const updateSelectedFilter = filter => ({
  type: 'UPDATE_SELECTED_FILTER',
  filter
});

export const updateSearchQuery = query => ({
  type: 'UPDATE_SEARCH_QUERY',
  query
});

export const updateError = errorMessage => ({
  type: 'UPDATE_ERROR',
  errorMessage
});

export const updateLoadingStatus = status => ({
  type: 'UPDATE_LOADING_STATUS',
  status
});
