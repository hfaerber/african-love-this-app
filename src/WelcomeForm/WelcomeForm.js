import React, { Component } from 'react';
import './WelcomeForm.css';

class WelcomeForm extends Component {
  constructor() {
    super();
    this.state = {
      name: ''
    }
  }

  handleChange = (event) => {
    this.setState({ name: event.target.value })
  }

  render() {
    return(
      <div>
        <h1>Welcome!</h1>
        <h2>This is African â™¥ This App</h2>
        <input type='text' className='input-name' placeholder='Enter your name'
          name='name' value={this.state.name} onChange={(event) => this.handleChange(event)} />
        <p>Why are you here {this.state.name ? `${this.state.name}` : ''}?</p>
        <div className='div-radio'>
          <label forhtml='bored' className='button-radio'>
            <input type='radio' id='bored'/>
            Bored
          </label>
          <label forhtml='egypt' className='button-radio'>
            <input type='radio' id='egypt'/>
            Found out Egypt was in Africa and it blew my mind
          </label>
          <label forhtml='learn' className='button-radio'>
            <input type='radio' id='learn'/>
            In college and wishing I could learn something I actually want to learn instead of my gen eds
          </label>
          <label forhtml='shame' className='button-radio'>
            <input type='radio' id='shame'/>
            Deeply ashamed of my lack of knowledge around World Geography and ready to change that
          </label>
        </div>

      </div>
    )
  }
}

export default WelcomeForm;

// within return i will render my NavBtn components once it is built and add onclick which navigates where you go
// and updates the redux store with the user name
