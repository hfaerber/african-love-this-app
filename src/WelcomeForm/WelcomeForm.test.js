import React from 'react';
import { WelcomeForm, mapDispatchToProps } from './WelcomeForm';
import { shallow } from 'enzyme';
import { updateUser } from '../actions';

describe('WelcomeForm', () => {
  let wrapper, mockEvent, mockUpdateUser;

  beforeEach(() => {
    mockUpdateUser = jest.fn();
    wrapper = shallow(<WelcomeForm
      updateUser={mockUpdateUser}
    />);
    mockEvent = { target: { value: 'test value' } };
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot when user has entered their name', () => {
    wrapper.setState({ name: 'Ben' });
    expect(wrapper).toMatchSnapshot();
  });

  it('should invoke handleChange on change', () => {
    wrapper.instance().handleChange = jest.fn();
    wrapper.find('.input-name').simulate('change', mockEvent)
    expect(wrapper.instance().handleChange).toHaveBeenCalledWith(mockEvent);
  });

  it('should update state when handleChange is invoked', () => {
    let initialState = { name: 'initial state value' };
    let expectedState = { name: 'test value' }
    wrapper.setState(initialState);
    wrapper.instance().handleChange(mockEvent);
    expect(wrapper.state()).toEqual(expectedState);
  });

  it('should invoke updateUser prop when handleChange is invoked', () => {
    let initialState = { name: 'initial state value' };
    let expectedState = { name: 'test value' }
    wrapper.setState(initialState);
    wrapper.instance().handleChange(mockEvent);
    expect(mockUpdateUser).toHaveBeenCalledWith('test value');
  });

  describe('mapDispatchToProps', () => {
    it('should call dispatch with an updateUser action when updateUser is called', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = updateUser('Ben');
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.updateUser('Ben');
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    })
  })
});
