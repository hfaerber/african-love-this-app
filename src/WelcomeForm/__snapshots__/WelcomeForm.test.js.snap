// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeForm should match the snapshot 1`] = `
<div>
  <h2
    className="welcome"
  >
    Welcome!
  </h2>
  <h3>
    This is 
    <span
      className="app-title"
    >
      African ♥ This App
    </span>
  </h3>
  <input
    className="input-name"
    name="name"
    onChange={[Function]}
    placeholder="Enter your name"
    type="text"
    value=""
  />
  <h4
    className="why-here"
  >
    Why are you here 
    ?
  </h4>
  <div
    className="div-radio"
  >
    <label
      className="label-radio"
      forhtml="bored"
    >
      <input
        className="button-radio"
        id="bored"
        type="radio"
      />
      Bored
    </label>
    <label
      className="label-radio"
      forhtml="egypt"
    >
      <input
        className="button-radio"
        id="egypt"
        type="radio"
      />
      Found out Egypt was in Africa and it blew my mind
    </label>
    <label
      className="label-radio"
      forhtml="learn"
    >
      <input
        className="button-radio"
        id="learn"
        type="radio"
      />
      In college and wishing I could learn something I actually want to learn instead of my gen eds
    </label>
    <label
      className="label-radio"
      forhtml="shame"
    >
      <input
        className="button-radio"
        id="shame"
        type="radio"
      />
      Deeply ashamed of my lack of knowledge around World Geography and ready to change that
    </label>
  </div>
  <div
    className="div-welcome-page-buttons"
  >
    <NavBtn
      link="/study"
      text="Study"
    />
    <NavBtn
      link="/quiz"
      text="Quiz"
    />
  </div>
</div>
`;

exports[`WelcomeForm should match the snapshot when user has entered their name 1`] = `
<div>
  <h2
    className="welcome"
  >
    Welcome!
  </h2>
  <h3>
    This is 
    <span
      className="app-title"
    >
      African ♥ This App
    </span>
  </h3>
  <input
    className="input-name"
    name="name"
    onChange={[Function]}
    placeholder="Enter your name"
    type="text"
    value="Ben"
  />
  <h4
    className="why-here"
  >
    Why are you here 
    Ben
    ?
  </h4>
  <div
    className="div-radio"
  >
    <label
      className="label-radio"
      forhtml="bored"
    >
      <input
        className="button-radio"
        id="bored"
        type="radio"
      />
      Bored
    </label>
    <label
      className="label-radio"
      forhtml="egypt"
    >
      <input
        className="button-radio"
        id="egypt"
        type="radio"
      />
      Found out Egypt was in Africa and it blew my mind
    </label>
    <label
      className="label-radio"
      forhtml="learn"
    >
      <input
        className="button-radio"
        id="learn"
        type="radio"
      />
      In college and wishing I could learn something I actually want to learn instead of my gen eds
    </label>
    <label
      className="label-radio"
      forhtml="shame"
    >
      <input
        className="button-radio"
        id="shame"
        type="radio"
      />
      Deeply ashamed of my lack of knowledge around World Geography and ready to change that
    </label>
  </div>
  <div
    className="div-welcome-page-buttons"
  >
    <NavBtn
      link="/study"
      text="Study"
    />
    <NavBtn
      link="/quiz"
      text="Quiz"
    />
  </div>
</div>
`;
